sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/events.scala

```diff
@@ -204,61 +204,64 @@ case class RenameFunctionEvent(
     newName: String)
   extends FunctionEvent
 
-trait PartitionEvent extends DatabaseEvent {
-  /**
-   * Name of the table that was touched.
-   */
-  val name: String
-}
-
+/**
+ * Event fired when a partition is created, dropped, altered or renamed.
+ */
+trait PartitionEvent extends TableEvent
 
 /**
  * Event fired before a partition is created.
  */
-case class CreatePartitionPreEvent(database: String, name: String,
-                                   parts: Seq[CatalogTablePartition]) extends PartitionEvent
+case class CreatePartitionPreEvent(
+    database: String,
+    name: String) extends PartitionEvent
```

Comment:
Ur? What I meant was we don't need to have the partition information at `PartitionEvent`. We need partition information in this event, don't we?

Reply:
For the indentation, shall we put `extends PartitionEvent` into a new line?

Reply:
Do you mean to create such a way:
```
 case class CreatePartitionPreEvent(
    database: String,
    name: String,
    parts: Seq[CatalogTablePartition]) extends PartitionEvent
```
but PartitionEvent remains empty ?

Reply:
> shall we put `extends PartitionEvent` into a new line

yes, I'll put it into a new line.